api/healthcheck/helper.go

```diff
@@ -0,0 +1,29 @@
+package healthcheck
+
+import (
+	"github.com/Peripli/service-manager/pkg/health"
+	"github.com/Peripli/service-manager/pkg/types"
+)
+
+func CheckPlatformsState(platforms []*types.Platform, fatal func(*types.Platform) bool) (map[string]*health.Health, int, int) {
+	details := make(map[string]*health.Health)
+	inactivePlatforms := 0
+	fatalInactivePlatforms := 0
+	for _, platform := range platforms {
+		healthObj := health.New().WithDetail("type", platform.Type)
+		if platform.Active {
+			healthObj = healthObj.WithStatus(health.StatusUp)
+		} else {
+			inactivePlatforms++
+			healthObj.WithStatus(health.StatusDown).WithDetail("since", platform.LastActive)
+			if fatal != nil && fatal(platform) {
+				fatalInactivePlatforms++
```

Comment:
don't we want an indicator in the details in case it is fatal?

Reply:
I didnt get quite the question. You have details always, what fatal function gives you is an option to decide based on some object which platform will consider as not healthy, for example, the old platform health indicator had a function plaform.type = cf , so only inactive platform of type cf could change the status of the healthcheck.